Polyfills/Canvas/Source/Canvas.cpp

```diff
@@ -99,31 +112,14 @@ namespace Babylon::Polyfills::Internal
         }
     }
 
-    struct TextureData final
-    {
-        ~TextureData()
-        {
-            if (bgfx::isValid(Handle))
-            {
-                bgfx::destroy(Handle);
-            }
-        }
-
-        bgfx::TextureHandle Handle{ bgfx::kInvalidHandle };
-        uint32_t Width{ 0 };
-        uint32_t Height{ 0 };
-        uint32_t Flags{ 0 };
-        uint8_t AnisotropicLevel{ 0 };
-    };
-
     Napi::Value NativeCanvas::GetCanvasTexture(const Napi::CallbackInfo& info)
     {
         assert(m_frameBufferHandle.idx != bgfx::kInvalidHandle);
-        auto data = new TextureData();
-        data->Handle = bgfx::getTexture(m_frameBufferHandle);
-        data->Width = m_width;
-        data->Height = m_height;
-        return Napi::External<TextureData>::New(info.Env(), data);
+        m_impl->m_textureData = std::make_unique<TextureData>();
```

Comment:
It looks like this gets deleted at the end of the function, but you are passing a pointer to the underlying data to the `Napi::External`. I think the default finalizer action for `Napi::External` is to delete the pointer, so it seems like it was correct the way it was, but maybe I'm missing something.

Reply:
It looks like `Napi::External` only deletes if you provide your own finalizer. This was likely leaking before.

Reply:
Hmm I wonder if we have leaks elsewhere…

Reply:
id bet we definitely have leaks elsewhere

Reply:
whatever was going on before, calling new just seems like a bad time for the TextureData when theres no subsequent delete call. To be fair though Cedric called out that this was leaking if i remember correctly.

Reply:
chatted with ryan offline. seems like a finalizer is what we should do here.